### PR TITLE
fix: change the type of the bool_and testcase to aggregate

### PR DESCRIPTION
**Pull request by EpsilonPrime**

### CODE REVIEW

File: tests/cases/boolean/bool_and.test [hunk 1]
```diff
@@ -1,4 +1,4 @@
-### SUBSTRAIT_SCALAR_TEST: v1.0
+### SUBSTRAIT_AGGREGATE_TEST: v1.0
```
Comment (scgkiran) [inline tests/cases/boolean/bool_and.test:1]: `bool.and` is a scalar function with variadic args. We should fix the tests.

Comment (EpsilonPrime) [reply tests/cases/boolean/bool_and.test:1]: This is bool.bool_and:  https://github.com/substrait-io/substrait/blob/main/extensions/functions_boolean.yaml#L118

Comment (scgkiran) [reply tests/cases/boolean/bool_and.test:1]: oh I missed the _ part.

**Review by scgkiran — APPROVED** [review]
LGTM

**Review by jacques-n — APPROVED** [review]

**Pull request merged**